app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt

```diff
@@ -130,71 +129,72 @@ fun EditMessageMenuItems(
-        }
-        add {
-            CompositionLocalProvider(LocalContentColor provides MaterialTheme.colorScheme.error) {
-                MenuBottomSheetItem(
-                    icon = {
-                        MenuItemIcon(
-                            id = R.drawable.ic_delete,
-                            contentDescription = stringResource(R.string.content_description_delete_the_message),
-                        )
-                    },
-                    title = stringResource(R.string.label_delete),
-                    onItemClick = onDeleteItemClick
-                )
+    if (message.expirationStatus is ExpirationStatus.Expirable) {
+        return EphemeralMessageEditMenuItems(
+            message = message,
+            onDetailsClick = onDetailsItemClick,
+            onDownloadAsset = onDownloadAssetClick,
+            onOpenAsset = onOpenAssetClick,
+            onDeleteMessage = onDeleteItemClick
+        )
+    } else {
+        return buildList {
+            if (isAvailable) {
+                add { ReactionOption(onReactionItemClick) }
+                add { MessageDetailsMenuOption(onDetailsItemClick) }
+                if (isCopyable) add { CopyItemMenuOption(onCopyItemClick) }
+                ReplyMessageOption(onReplyItemClick)
+                if (isAssetMessage) add { DownloadAssetExternallyOption(onDownloadAssetClick) }
+                if (isGenericAsset) add { OpenAssetExternallyOption(onOpenAssetClick) }
+                if (isEditable) { add { EditMessageMenuOption(onEditItemClick) } }
+                if (isAssetMessage) { add { ShareAssetMenuOption(onShareAsset) } }
+                add { DeleteItemMenuOption(onDeleteItemClick) }
             }
         }
     }
 }
 
+@Composable
+fun CopyItemMenuOption(onCopyItemClick: () -> Unit) {
+    MenuBottomSheetItem(
+        icon = {
+            MenuItemIcon(
+                id = R.drawable.ic_copy,
+                contentDescription = stringResource(R.string.content_description_copy_the_message),
+            )
+        },
+        title = stringResource(R.string.label_copy),
+        onItemClick = onCopyItemClick
+    )
+}
+
+
+@Composable
+fun ShareAssetMenuOption(onShareAsset: () -> Unit) {
+    MenuBottomSheetItem(
+        icon = {
+            MenuItemIcon(
+                id = R.drawable.ic_share_file,
+                contentDescription = stringResource(R.string.content_description_share_the_file),
+            )
+        },
+        title = stringResource(R.string.label_share),
+        onItemClick = onShareAsset
+    )
+}
+
+@Composable
+private fun EditMessageMenuOption(onEditItemClick: () -> Unit) {
+    MenuBottomSheetItem(
+        icon = {
+            MenuItemIcon(
+                id = R.drawable.ic_edit,
+                contentDescription = stringResource(R.string.content_description_edit_the_message)
+            )
+        },
+        title = stringResource(R.string.label_edit),
+        onItemClick = onEditItemClick
+    )
+}
```

Comment:
I would still move them to separate files for consistency with the other options. But just personal preference.

Reply:
It's not about solving anything, is about treating all the edit options in the same way. Since we have most of these options already extracted to separated files, let's do the same with each of these ones to allow easier reusability in the future and increase concern separation bitte.